modules/replication/dockerimage-replication/replicate_charts.py

```diff
@@ -0,0 +1,135 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+import json
+import os
+import sys
+import time
+from typing import Any, Dict
+
+from replication.ecr.ecr_utils import ECRUtils
+from replication.logging import logger
+from replication.utils import export_results, get_credentials, run_command
+
+successful_replication = []
+failed_replication = []
+
+aws_account_id = os.getenv("AWS_ACCOUNT_ID")
+aws_region = os.getenv("AWS_DEFAULT_REGION")
+aws_partition = os.getenv("AWS_PARTITION", "aws")
+aws_domain = "amazonaws.com" if aws_partition == "aws" else "amazonaws.com.cn"
+
+repo_secret = os.getenv("SEEDFARMER_PARAMETER_HELM_REPO_SECRET_NAME", None)
+repo_key = os.getenv("SEEDFARMER_PARAMETER_HELM_REPO_SECRET_KEY", None)
+
+
+def process_chart(ecr_utils: ECRUtils, chart_key: str, chart_data: Dict[str, Any]) -> None:
+    """Process a single chart from the JSON document."""
+    logger.info(f"Processing chart: {chart_key} ")
+
+    # Extract Helm and repository details
+    helm = chart_data["helm"]
+    name = helm["name"]
+    version = helm["version"]
+    src_repository = helm["srcRepository"]
+    target_repository = helm["repository"]
+
+    # Prepare for push
+    chart_package = f"{name}-{version}.tgz"
+    ecr_repo_name = f"{target_repository.replace('oci://', '').split('/', 1)[-1]}"
+    ecr_repo_target = target_repository.replace(f"/{name}", "")
+
+    if ecr_utils.image_exists(ecr_repo_name, version):
+        logger.info(f"Chart {name} with version {version} already exists in {ecr_repo_name}. Skipping.")
+        successful_replication.append(name)
+        return
```

Comment:
Is this return expected? This is going to stop chart replication after the first existing chart that already exists. Same on L55

Reply:
I believe that it moves onto the next chart in the list (at least the testing has indicated that this is the case)

Reply:
You're right, my bad - looks good